lib/tasks/deleted_objects.rake

```diff
@@ -11,4 +11,16 @@ namespace :deleted_objects do
       sleep(0.5) unless Rails.env.test?
     end
   end
+
+  desc 'Destroy all the service objects and those whose owner is also a deleted object. Objects destroyed at least 1 week ago.'
+  task :destroy_stale_or_service_objects => :environment do
+    stale_or_service_objects = DeletedObject.where.has do
+      ((id.in DeletedObject.deleted_owner.select(:id)) | (object_type == Service.name)) \
+      & (created_at <= 1.week.ago)
+    end
+    stale_or_service_objects.find_in_batches(batch_size: 500) do |records|
+      DeletedObject.where(id: records.map(&:id)).destroy_all
```

Comment:
This will generate a SQL ÌN` condition with 500 values. Is this all right (performance-wise)?

Reply:
Probably so. We should test in preview to be sure, but the SQL generated in the rake task right above is with batch size 300 and was executed beyond smoothly back then, so it looks good 😄